xtc/src/ledger.rs

```diff
@@ -97,11 +181,38 @@ static InsufficientBalanceError: ErrorDetails = ErrorDetails {
     msg: "Insufficient Balance",
     code: APIError::InsufficientBalance,
 };
+
+static InsufficientAllowanceError: ErrorDetails = ErrorDetails {
+    msg: "Insufficient Allowance",
+    code: APIError::InsufficientAllowance,
+};
+
 static UnknownError: ErrorDetails = ErrorDetails {
     msg: "Unknown",
     code: APIError::Unknown,
 };
 
+// Disabled as the `name` clashes with a similar method in the cycles wallet
+// #[query]
+// fn name() -> &'static str {
+//     meta().name
+// }
+
+#[query]
+fn symbol() -> &'static str {
+    meta().symbol
+}
+
+#[query]
+fn decimal() -> u8 {
+    meta().decimal
+}
+
+#[query]
+fn totalSupply() -> Nat {
+    StatsData::get().supply
+}
```

Comment:
I'd move this methods to `meta.rs` unless you want to change the directory structure as well.